src/components/common/BuyCryptoButton/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { usePathname, useSearchParams } from 'next/navigation'
+import Link, { type LinkProps } from 'next/link'
+import { Alert, Box, Button, ButtonBase, Typography } from '@mui/material'
+import AddIcon from '@mui/icons-material/Add'
+import { SafeAppsTag } from '@/config/constants'
+import { AppRoutes } from '@/config/routes'
+import { useRemoteSafeApps } from '@/hooks/safe-apps/useRemoteSafeApps'
+import madProps from '@/utils/mad-props'
+import { type ReactNode, useMemo } from 'react'
+import Track from '../Track'
+import { OVERVIEW_EVENTS } from '@/services/analytics'
+import RampLogo from '@/public/images/common/ramp_logo.svg'
+import css from './styles.module.css'
+import ChevronRightRoundedIcon from '@mui/icons-material/ChevronRightRounded'
+
+const useOnrampAppUrl = (): string | undefined => {
+  const [onrampApps] = useRemoteSafeApps(SafeAppsTag.ONRAMP)
```

Comment:
This will cause a fetch of the safe app list, so please make sure it's not triggered unnecessarily.
On the Dashboard, we already fetch that list anyway, so it should be fine, but on other pages it will cause an extra request.